crates/iroha_torii/src/routing.rs

```diff
@@ -103,110 +97,143 @@ pub async fn handle_post_configuration(
     Ok((StatusCode::ACCEPTED, ()))
 }
 
-#[iroha_futures::telemetry_future]
-pub async fn handle_blocks_stream(kura: Arc<Kura>, mut stream: WebSocket) -> eyre::Result<()> {
-    let BlockSubscriptionRequest(mut from_height) =
-        Stream::<BlockSubscriptionRequest>::recv(&mut stream).await?;
-
-    let mut interval = tokio::time::interval(std::time::Duration::from_millis(10));
-    loop {
-        // FIXME: cleanup.
-
-        tokio::select! {
-            // This branch catches `Close` and unexpected messages
-            closed = async {
-                while let Some(message) = stream.try_next().await? {
-                    if message.is_close() {
-                        return Ok(());
-                    }
-                    iroha_logger::warn!(?message, "Unexpected message received");
-                }
-                eyre::bail!("Can't receive close message")
-            } => {
-                match closed {
-                    Ok(()) =>  {
-                        return stream.close().await.map_err(Into::into);
-                    }
-                    Err(err) => return Err(err)
-                }
+pub mod block {
+    //! Blocks stream handler
+
+    use stream::WebSocketScale;
+
+    use super::*;
+    use crate::block;
+
+    /// Type for any error during blocks streaming
+    #[derive(Debug, displaydoc::Display, thiserror::Error)]
+    enum Error {
+        /// Block consumption resulted in an error: {_0}
+        Consumer(#[from] Box<block::Error>),
+        /// Event reception error
+        Event(#[from] tokio::sync::broadcast::error::RecvError),
```

Comment:
Is this error variant used anywhere? I don't see any broadcast channels being used for blocks..

Reply:
Indeed, removed.